server/certstore/certstore_other.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022-2023 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !windows
+
+package certstore
+
+import (
+	"crypto"
+	"crypto/tls"
+	"io"
+)
+
+var _ = MATCHBYEMPTY
+
+// otherKey implements crypto.Signer and crypto.Decrypter to satisfy linter on platforms that don't implement certstore
+type otherKey struct{}
+
+func TLSConfig(certStore StoreType, certMatchBy MatchByType, certMatch string, config *tls.Config) error {
+	_, _, _, _ = certStore, certMatchBy, certMatch, config
+	return ErrOSNotCompatCertStore
+}
+
+// Public always returns nil public key since this is a stub on non-supported platform
+func (k otherKey) Public() crypto.PublicKey {
+	return nil
+}
+
+// Sign always returns a nil signature since this is a stub on non-supported platform
+func (k otherKey) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error) {
+	_, _, _ = rand, digest, opts
```

Comment:
What is this?

Reply:
Stub implementation for non-Windows ties back to L26 comment and root avoidance of creating a Windows platform copy of `opts.go` since that is such a _special_ source code file to keep maintained.

L40-41 in particular could be alternately phrased as:

```golang
func TLSConfig(_ StoreType, _ MatchByType, _ string, _ *tls.Config) error {
	// _, _, _, _ = certStore, certMatchBy, certMatch, config
	return ErrOSNotCompatCertStore
}
```

Reply:
Whoops.  Gave an alternate phrasing example for L29-32, not L40-41 but same thing.  Linters complaining about unused variables...

Reply:
For more background, this code in `opts.go` is cross-platform:

https://github.com/nats-io/nats-server/blob/f854e95c1101b3894dddb7044a7eb3962136c866/server/opts.go#L4342

The options and interface potentially directly usable if we want to tie into other 3rd-party stores in future...

Reply:
I think the better pattern is to change the args to the functions as `_`